widget/embedded/src/utils/common.ts

```diff
@@ -32,8 +41,180 @@ export function containsText(text: string, searchText: string): boolean {
   return text.toLowerCase().indexOf(searchText.toLowerCase()) > -1;
 }
 
+export function exactText(text: string, searchText: string): boolean {
+  return text.toLowerCase() === searchText.toLowerCase();
+}
+
+export function startWithText(text: string, searchText: string): boolean {
+  return text.toLowerCase().startsWith(searchText.toLowerCase());
+}
+
 export const getContainer = () =>
   document.getElementById(WIDGET_UI_ID.SWAP_BOX_ID) as HTMLElement;
 
 export const getExpanded = () =>
   document.getElementById(WIDGET_UI_ID.EXPANDED_BOX_ID) as HTMLElement;
+
+function compareExactMatchText(
+  searchFor: string,
+  textA?: string | null,
```

Comment:
why `textA` and `textB` can be `undefined`? if they are undefined, why we are comparing?
Same comment applies to other function in this file.

Reply:
Because it can be **null** in some cases.
And I didn't want these conditions to increase the complexity of where this function is called.
The comparison between **null** and **string** is also handled in this mode and priority is given to the string.


In this example, the token name may be **null**:
```
  // 2: Check the order to exact match the token name
  if (searchFor.length >= MIN_LENGTH_TOKEN_NAME_EXACT_MATCH) {
    const tokenNameExactMatchCompare = compareExactMatchText(
      searchFor,
      tokenA.name,
      tokenB.name
    );
    if (tokenNameExactMatchCompare) {
      return tokenNameExactMatchCompare;
    }
  }
```

Reply:
I meant `?` case. you accept 3 inputs right now, `string | null | undefined`. By removing `?`, it will be ``string | null`.

Reply:
Fixed